source/core/Ky.ts

```diff
@@ -86,9 +86,13 @@ export class Ky {
 		const isRetriableMethod = ky._options.retry.methods.includes(ky.request.method.toLowerCase());
 		const result = (isRetriableMethod ? ky._retry(function_) : function_())
 			.finally(async () => {
-				// Now that we know a retry is not needed, close the ReadableStream of the cloned request.
-				if (!ky.request.bodyUsed) {
-					await ky.request.body?.cancel();
+				const originalRequest = ky._originalRequest;
+
+				// Cancel both the original and cloned request bodies to prevent hanging.
+				if (originalRequest && !originalRequest.bodyUsed && !ky.request.bodyUsed) {
```

Comment:
Is there a situation where one request body _does_ get used but we still need to cancel the other one? Maybe we shouldn't lump these together?

In fact, the reason why the code here originally only cancelled the clone (`ky.request`) is because our expectation was that the original (`ky._originalRequest`) _would_ be consumed by fetch during the normal course of trying to send the request. So if I'm not mistaken, there will often be a situation where `originalRequest.bodyUsed` is `true` because the request was sent, but `ky.request.bodyUsed` is `false` because a retry was not performed. Don't we still need to call `ky.request.body.cancel()` in that case?